packages/next/cli/next-dev.ts

```diff
@@ -73,12 +73,16 @@ const nextDev: cliCommand = (argv) => {
 
   const port =
     args['--port'] || (process.env.PORT && parseInt(process.env.PORT)) || 3000
-  const host = args['--hostname'] || '0.0.0.0'
-  const appUrl = `http://${host === '0.0.0.0' ? 'localhost' : host}:${port}`
+
+  // We do not set a default host value here to prevent breaking
+  // some set-ups that rely on listening on other interfaces
+  const host = args['--hostname']
+  const ipPort = `${!host ? 'localhost' : host}:${port}`
```

Comment:
```suggestion
  const ipPort = `${host || 'localhost'}:${port}`
```

I think this is equivalent